src/core/drive/page_snapshot.ts

```diff
@@ -30,9 +30,9 @@ export class PageSnapshot extends Snapshot<HTMLBodyElement> {
     const clonedSelectElements = clonedElement.querySelectorAll("select")
 
     for (const [index, source] of selectElements.entries()) {
-      for (const [optionIndex, option] of Array.from(source.options).entries()) {
-        clonedSelectElements[index].options[optionIndex].selected = option.selected
-      }
+      const clone = clonedSelectElements[index]
+      for (const option of clone.selectedOptions) option.selected = false
+      for (const option of source.selectedOptions) clone.options[option.index].selected = true
```

Comment:
Yes, this change is also significantly faster than the change I suggested in the previous PR.

Overall in the HTML page I tested:
- original: 0.5ms to 1.0ms
- updated: 0.07ms to 0.1ms